### PR TITLE
Add default failure message for expect_same expectations

### PR DESCRIPTION
**Pull request by nelsonkopliku**
This PR takes care of providing a default failure message for failing expect_same statements.

### CODE REVIEW

**Review by arbulu89 — APPROVED** [review]
Nice!

PD:
Now I saw the code it would've been maybe easier to put the detaulf value in the catalog code (https://github.com/trento-project/wanda/blob/main/lib/wanda/catalog.ex#L121):
```
failure_message: Map.get(expectation, "failure_message", @default_failure_message)
```
It was just me thinking aloud, don't change now hehe

File: test/wanda/executions/evaluation_test.exs [hunk 1]
```diff
@@ -1363,6 +1363,127 @@ defmodule Wanda.Executions.EvaluationTest do
...
+                         %ExpectationEvaluation{
+                           name: ^another_expectation_name,
+                           return_value: false,
+                           type: :expect_same
+                         }
+                       ],
+                       facts: ^incorrect_facts
+                     }
+                   ],
+                   expectation_results: [
```
[the remark this answers is not in the document]
Comment (fabriziosestito) [reply test/wanda/executions/evaluation_test.exs:1466]: +1

**Comment by fabriziosestito** [comment]
> Nice!
> 
> PD: Now I saw the code it would've been maybe easier to put the detaulf value in the catalog code (https://github.com/trento-project/wanda/blob/main/lib/wanda/catalog.ex#L121):
> 
> ```
> failure_message: Map.get(expectation, "failure_message", @default_failure_message)
> ```
> 
> It was just me thinking aloud, don't change now hehe

@arbulu89 My 2cents, I keep this in the evaluation module, since it is part of the domain logic. The catalog itself should just abstract the catalog loading without introducing logic. 
Also, `@default_failure_message` is used elsewhere in the evaluation core, so it seems to be the right place for it.

**Review by fabriziosestito — APPROVED** [review]
LGTM
:shipit:  once the test assertion is fixed

**Review by dottorblaster — APPROVED** [review]
LGTM

**Pull request merged**